### ISSUE TITLE
Wrong cyclomatic complexity calculation

### ISSUE THREAD

**Issue by spyric** [nunomaduro/phpinsights#134]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| Library version  | 1.3.1

I found some strange issue with calculating cyclomatic complexity.

It seems to be bug in Analyser class that count any `?` as ternary condition and even optional typehint increase class cyclomatic complexity. For example, next code has complexity equal to 6

````
abstract class User  {
    protected function setId(?UserId $id): void
    {
        $this->id = $id;
    }

    protected function setName(?FullName $name): void
    {
        $this->name = $name;
    }

    protected function setEmail(?Email $email): void
    {
        $this->email = $email;
    }

    protected function setPassword(?Password $password): void
    {
        $this->password = $password;
    }

    protected function setAvatar(?Image $avatar): void
    {
       $this->avatar = $avatar;
    }
}
````
--

**Comment by nunomaduro** [comment]
Can you try to debug it ?

**Comment by spyric** [comment]
I think it caused by this condition: https://github.com/nunomaduro/phpinsights/blob/c1a99b3/src/Domain/Analyser.php#L104

I will try to understand how parser works and prevent increasing complexity value because of `?` in function declaration

**Comment by spyric** [comment]
Fixed by #150
